File: ArtSiteApi/ArtSite.Data/Models/ArtworkModel.cs
using ArtSite.Data.Enums;

namespace ArtSite.Data.Models;

public class ArtworkModel : BaseModel{
    public byte[] Description { get; set; }
    public byte[] File { get; set; }
    public string Extension { get; set; }
    public NSFWRating NsfwRating { get; set; }
    public GoreRating GoreRating { get; set; }
    public HashSet<ArtistModel> Artists { get; set; }
    public HashSet<CharacterModel> Characters { get; set; }
    public HashSet<TagModel> Tags { get; set; }
};

File: ArtSiteWeb/artsite-web/src/network/models/artworkModel.ts
import { Rating } from "../enums/rating";
import { ArtistModel } from "./artistModel";
import { BaseModel } from "./baseModel";
import { CharacterModel } from "./characterModel";

export interface ArtworkModel extends BaseModel {
    description: string;
    file: string;
    extension: string;
    rating: Rating;
    artists: ArtistModel[];
    characters: CharacterModel[];
}
